### ISSUE TITLE
Mod is invalid?

### ISSUE THREAD

**Issue by cauliflower69** [Quetzi/Whitelister#3]
Using the latest version for 1.14 and we are having an issue. 

[08Dec2019 15:27:37.233] [main/WARN] [net.minecraftforge.fml.loading.moddiscovery.ModFileParser/LOADING]: Mod file Mod File: /home/servers/atm4/Minecraft/mods/Whitelister-1.14.4-4.1.14.jar is missing mods.toml file
[08Dec2019 15:27:37.233] [main/WARN] [net.minecraftforge.fml.loading.moddiscovery.ModDiscoverer/SCAN]: File /home/servers/atm4/Minecraft/mods/Whitelister-1.14.4-4.1.14.jar has been ignored - it is invalid

**Comment by Quetzi** [comment]
Fixed in version 4.1.16

**Comment by cauliflower69** [comment]
Same issue with the latest version unfortunatley

[11Dec2019 12:10:30.010] [main/WARN] [net.minecraftforge.fml.loading.moddiscovery.ModFileParser/LOADING]: Mod file Mod File: /home/servers/test/Minecraft/mods/Whitelister-1.14.4-4.1.16.jar is missing mods.toml file
[11Dec2019 12:10:30.011] [main/WARN] [net.minecraftforge.fml.loading.moddiscovery.ModDiscoverer/SCAN]: File /home/servers/test/Minecraft/mods/Whitelister-1.14.4-4.1.16.jar has been ignored - it is invalid


[11Dec2019 12:10:48.661] [Server thread/WARN] [net.minecraftforge.fml.server.ServerModLoader/LOADING]: File Whitelister-1.14.4-4.1.16.jar is not a valid mod file

**Comment by Quetzi** [comment]
Should actually be fixed in 4.1.19

**Comment by cauliflower69** [comment]
Crashes on startup https://pastebin.com/fCJRd33u

Here is the log file https://pastebin.com/DNBLG242

**Comment by Quetzi** [comment]
4.1.20 should resolve that crash

**Comment by cauliflower69** [comment]
https://pastebin.com/WFktYbeW - Crash report

https://pastebin.com/Wk8mNcBX - logs

Sorry to be a pita

**Comment by Quetzi** [comment]
Looks like a separate issue with the new config system, try build 22

**Comment by cauliflower69** [comment]
https://pastebin.com/vSuYieCP - Crash

https://pastebin.com/ygDRJjV2 - logs

**Comment by Quetzi** [comment]
There is a new build, 23 which may or may not fix this crash.

**Comment by cauliflower69** [comment]
Must be waiting on approval as I can't grab it from curseforge but it is showing as available. I'll report back with the results.

Thanks for your hard work

**Comment by cauliflower69** [comment]
https://pastebin.com/jZdnY39u - Crash

https://pastebin.com/kfP0e54j - Log

**Comment by Quetzi** [comment]
Ok, looks like the latest build is working properly. Your new crash is for com.electronwill.nightconfig whichever mod that belongs to. I'll update the latest build to release, thanks for your help with this :)

**Comment by cauliflower69** [comment]
Yours is the only mod loaded

Edit: There is no cfg/toml file created.

Np happy to help

**Comment by Quetzi** [comment]
It's a server config so it gets saved in the world directory.

**Comment by cauliflower69** [comment]
Ahhh TIL! I was looking for the Forge remove erroring entities settings since the move to 1.1.4

The file whitelister-server.toml is empty unlike the previous versions is this intended?

**Comment by Quetzi** [comment]
`[11Dec2019 22:29:22.310] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Configuration file ./world/serverconfig/whitelister-server.toml is not correct. Correcting
[11Dec2019 22:29:22.311] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key settings was corrected from null to SimpleCommentedConfig:{}
[11Dec2019 22:29:22.312] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key settings.isEnabled was corrected from null to true
[11Dec2019 22:29:22.312] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key settings.urlList was corrected from null to [http://example.com/whitelist.txt, http://example.com/whitelist2.txt]
[11Dec2019 22:29:22.312] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key settings.checkInterval was corrected from null to 10
[11Dec2019 22:29:22.312] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key settings.kickMessage was corrected from null to You are not on the whitelist`

That's the extract from your log where the config is being populated with the correct values. It's likely that the subsequent crash caused them to not be saved.

**Comment by cauliflower69** [comment]
I'll launch again with only that mod again and a new world. Thanks again

**Comment by cauliflower69** [comment]
https://pastebin.com/sQbRayhh - Crash

https://pastebin.com/cSdH4kXF - Log

**Comment by Quetzi** [comment]
That is the same crash, from the other mod.
`[11Dec2019 23:09:05.684] [Server thread/ERROR] [net.minecraft.server.MinecraftServer/]: Encountered an unexpected exception
com.electronwill.nightconfig.core.io.WritingException: Unsupported value type: class [Ljava.lang.String;`

**Comment by cauliflower69** [comment]
There is no other mod loaded. Yours is the only one in the mod folder. There are no datapacks also.

**Comment by Quetzi** [comment]
I'm now working on the assumption that the new forge config system cannot cope with string arrays, the latest build (25) will now only accept a single url.

**Comment by cauliflower69** [comment]
The mod is now generating the full cfg file and I have amended it to use a test whitelist file and have set the server.properties file to whitelist=false.


[settings]
	#Time between checks in minutes
	#Range: 1 ~ 32000
	checkInterval = 1
	#Enable whitelister
	isEnabled = true
	#Comma separated url List of plain text files
	url = "https://requiemofapickaxe.com/~servers/whitelist/whitelisttest.txt"
	#Kick Message
	kickMessage = "You are not on the whitelist"

I can however log in with my username Cauli_

https://pastebin.com/ayiLsNQr - Logs

**Comment by Quetzi** [comment]
4.1.26 should fix that crash.

**Comment by cauliflower69** [comment]
https://requiemofapickaxe.com/~servers/test/latest.log = log
